Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -812,9 +805,9 @@ public struct ObjectId: BSONValue, Equatable, CustomStringConvertible, Codable {
 
     /// Initializes an `ObjectId` from the provided hex `String`. Returns `nil` if the string is not a valid ObjectId.
     /// - SeeAlso: https://github.com/mongodb/specifications/blob/master/source/objectid.rst
-    public init?(_ hex: String) {
+    public init(_ hex: String) throws {
         guard bson_oid_is_valid(hex, hex.utf8.count) else {
-            return nil
+            throw InternalError(message: "Cannot create ObjectId from \(hex)")
```

Comment:
I think this should be an `InvalidArgumentError`, as it means the user gave us invalid input.